--- bash-4.2/execute_cmd.c	2013-11-30 18:38:27.238855662 +0100
+++ bash-4.2/execute_cmd.c	2013-11-30 18:36:52.655658242 +0100
@@ -2202,7 +2202,12 @@
   /* If the `lastpipe' option is set with shopt, and job control is not
      enabled, execute the last element of non-async pipelines in the
      current shell environment. */
+#if defined (JOB_CONTROL)
   if (lastpipe_opt && job_control == 0 && asynchronous == 0 && pipe_out == NO_PIPE && prev > 0)
+#else
+  int job_control=0;
+  if (lastpipe_opt && job_control == 0 && asynchronous == 0 && pipe_out == NO_PIPE && prev > 0)
+#endif
     {
       lstdin = move_to_high_fd (0, 0, 255);
       if (lstdin > 0)
