--- neutrino-hd2-exp/src/driver/vfd.cpp	2014-02-09 17:01:06.452378461 +0100
+++ neutrino-hd2-exp/src/driver/vfd.cpp	2014-02-09 17:06:26.551678643 +0100
@@ -47,15 +47,11 @@
 
 
 #if defined (__sh__)
-#if defined (PLATFORM_SPARK7162)
 static struct aotom_ioctl_data aotom_data;
-#endif
 
-#if defined (PLATFORM_KATHREIN) || defined (PLATFORM_SPARK7162)
 static bool usb_icon = false;
 static bool timer_icon = false;
 static bool hdd_icon = false;
-#endif
 
 //konfetti: let us share the device with evremote and fp_control
 //it does currently not support more than one user (see e.g. micom)
@@ -337,12 +333,6 @@
 			{
 				hour = t->tm_hour;
 				minute = t->tm_min;
-#if defined (PLATFORM_KATHREIN)							/* time and date at kathrein because 16 character vfd	*/
-				strftime(timestr, 20, "%H:%M - %d.%m.%y", t);
-#elif !defined(PLATFORM_SPARK7162) && !defined (PLATFORM_KATHREIN)		/* no time at spark7162 because clock integrated	*/
- 				strftime(timestr, 20, "%H:%M", t);
-#endif				
-				ShowText(timestr);
 			}
 		} 
 	}
@@ -447,27 +437,21 @@
 	{
 		case MODE_TVRADIO:	
 			showServicename(servicename);
-#if !defined(PLATFORM_SPARK7162)			
-			ShowIcon(VFD_ICON_MP3, false);	        // NOTE: @dbo  //ICON_MP3 and ICON_DOLBY switched in infoviewer 
-#endif			
 	
 #if defined (PLATFORM_KATHREIN)
 			ShowIcon(VFD_ICON_USB, usb_icon);	
 			ShowIcon(VFD_ICON_HDD, hdd_icon);	
-#elif defined(PLATFORM_SPARK7162)
+#endif			
 			ShowIcon(VFD_ICON_USB, usb_icon);	
 			ShowDiskLevel();
 			ShowIcon(VFD_ICON_STANDBY, false);	
-#endif
 			showclock = true;
 			//showTime();      /* "showclock = true;" implies that "showTime();" does a "displayUpdate();" */			
 			break;
 
 		case MODE_AUDIO:
 		{
-#if defined(PLATFORM_SPARK7162)
 			ShowIcon(VFD_ICON_AC3, false);			
-#endif		  
 			ShowIcon(VFD_ICON_MP3, true);			
 			ShowIcon(VFD_ICON_TV, false);			
 			showAudioPlayMode(AUDIO_MODE_STOP);			
@@ -498,18 +482,18 @@
 			//Clear();
 			/* clear all symbols */
 			ClearIcons();
-#if defined(PLATFORM_SPARK7162)
+
 			ShowIcon(VFD_ICON_CLOCK, timer_icon);	
-#endif			
+		
 			showclock = false;
 			break;
 
 		case MODE_STANDBY:
 			ShowIcon(VFD_ICON_TV, false);
 			ClearIcons();
-#if defined(PLATFORM_SPARK7162)
+
 			ShowIcon(VFD_ICON_STANDBY, true);	
-#endif			
+		
 			showclock = true;
 			showTime(true);      	/* "showclock = true;" implies that "showTime();" does a "displayUpdate();" */
 						/* "showTime()" clears the whole lcd in MODE_STANDBY */
@@ -683,35 +667,16 @@
 	if(!has_lcd || is4digits) 
 		return;
 
-#if defined (__sh__)	
-#if defined(PLATFORM_SPARK7162)		/* using one command for switching off all Icons*/	 
+#if defined (__sh__)		 
 	openDevice();
 	aotom_data.u.icon.icon_nr = SPARK_ICON_ALL;
 	aotom_data.u.icon.on = 0;
 	if (ioctl(fd, VFDICONDISPLAYONOFF, &aotom_data) <0)
 		perror("VFDICONDISPLAYONOFF");
 	closeDevice();
-#else
-	int i;
-	struct vfd_ioctl_data data;
-	
-	openDevice();
-	
-	for(i = 0; i <= 15; i++)
-	{
-		data.data[0] = i;
-		data.data[4] = 0;
-		
-		if( ioctl(fd, VFDICONDISPLAYONOFF, &data) < 0)
-			perror("VFDICONDISPLAYONOFF");
-	}
-	
-	closeDevice();
-#endif
 #endif
 }
-
-#if defined(PLATFORM_SPARK7162)			/* only for Spark7162 STB's which Display has a HDD Level indicator */	 
+		/* only for Spark7162 STB's which Display has a HDD Level indicator */	 
 void CVFD::ShowDiskLevel()
 {
 	int hdd_icons[9] = {24, 23, 21, 20, 19, 18, 17, 16, 22};
@@ -743,7 +708,7 @@
 
 	}
 }
-#endif
+
 
 void CVFD::ShowIcon(vfd_icon icon, bool show)
 {
@@ -753,7 +718,6 @@
 	dprintf(DEBUG_DEBUG, "CVFD::ShowIcon %s %x\n", show ? "show" : "hide", (int) icon);
 
 #if defined (__sh__)
-#if defined (PLATFORM_KATHREIN) || defined(PLATFORM_SPARK7162)
 	switch (icon)
 	{
 		case VFD_ICON_USB:
@@ -770,31 +734,14 @@
 		default:
 			break;
 	}
-#endif
 
 	openDevice();
-#if defined(PLATFORM_SPARK7162)
 	aotom_data.u.icon.icon_nr = icon;
 	aotom_data.u.icon.on = show ? 1 : 0;
 	
 	if (ioctl(fd, VFDICONDISPLAYONOFF, &aotom_data) <0)
 		perror("VFDICONDISPLAYONOFF");	
-#else
-#if defined (PLATFORM_KATHREIN)
-	if (icon == 17)				/* returning because not existing icon at ufs910 */
-	{
-		closeDevice();	
-		return;
-	}
-#endif	
-	struct vfd_ioctl_data data;
 
-	data.data[0] = (icon - 1) & 0x1F;
-	data.data[4] = show ? 1 : 0;
-	
-	if( ioctl(fd, VFDICONDISPLAYONOFF, &data) < 0)
-		perror("VFDICONDISPLAYONOFF");
-#endif	
 	closeDevice();
 #elif defined (PLATFORM_COOLSTREAM)
 	int ret = ioctl(fd, show ? IOC_VFD_SET_ICON : IOC_VFD_CLEAR_ICON, icon);
--- neutrino-hd2-exp/src/driver/vfd.h	2014-01-23 16:07:07.825703686 +0100
+++ neutrino-hd2-exp/src/driver/vfd.h	2014-02-09 17:01:11.904192435 +0100
@@ -38,7 +38,6 @@
 #endif
 
 
-#if defined(PLATFORM_SPARK7162)
 /* SAB Triple VFD Icons								*/
 /* taken from aotom_main.h, with these numbers the Icons on Spark7162		*/
 /* will work right ( hopefully all Models )					*/
@@ -131,7 +130,6 @@
 	VFD_ICON_STANDBY	= VFD_SPARK_CYCLE	
 } vfd_icon;
 
-#endif //spark7162
 
 #if defined(PLATFORM_KATHREIN)
 typedef enum
@@ -179,66 +177,8 @@
 } vfd_icon;
 #endif //ufs910, ufs913
 
-#if !defined(PLATFORM_KATHREIN) && !defined(PLATFORM_SPARK7162) && !defined (PLATFORM_COOLSTREAM)
-// duckbox
-// token from micom
-enum {
-	ICON_MIN,             // 0x00
-	ICON_STANDBY,
-	ICON_SAT,
-	ICON_REC,
-	ICON_TIMESHIFT,
-	ICON_TIMER,           // 0x05
-	ICON_HD,
-	ICON_USB,
-	ICON_SCRAMBLED,
-	ICON_DOLBY,
-	ICON_MUTE,            // 0x0a
-	ICON_TUNER1,
-	ICON_TUNER2,
-	ICON_MP3,
-	ICON_REPEAT,
-	ICON_Play,            // 0x0f
-	ICON_TER,            
-	ICON_FILE,
-	ICON_480i,
-	ICON_480p,
-	ICON_576i,            // 0x14
-	ICON_576p,
-	ICON_720p,
-	ICON_1080i,
-	ICON_1080p,
-	ICON_Play_1,          // 0x19 
-	ICON_RADIO,   
-	ICON_TV,      
-	ICON_PAUSE,   
-	ICON_MAX
-};
 
-// neutrino common
-typedef enum
-{
-	VFD_ICON_MUTE		= ICON_MUTE,
-	VFD_ICON_DOLBY		= ICON_DOLBY,
-	VFD_ICON_POWER		= ICON_STANDBY,
-	VFD_ICON_TIMESHIFT	= ICON_REC,
-	VFD_ICON_TV		= ICON_TV,
-	VFD_ICON_RADIO		= ICON_TV,
-	VFD_ICON_HD		= ICON_HD,
-	VFD_ICON_1080P		= ICON_1080p,
-	VFD_ICON_1080I		= ICON_1080i,
-	VFD_ICON_720P		= ICON_720p,
-	VFD_ICON_480P		= ICON_480p,
-	VFD_ICON_480I		= ICON_480i,
-	VFD_ICON_USB		= ICON_USB,
-	VFD_ICON_MP3		= ICON_MP3,
-	VFD_ICON_PLAY		= ICON_Play,
-	VFD_ICON_PAUSE		= ICON_PAUSE,
-	VFD_ICON_LOCK 		= ICON_SCRAMBLED
-} vfd_icon;
-#endif  //common
 
-#if defined (__sh__)
 #define VFDBRIGHTNESS         0xc0425a03
 #define VFDPWRLED             0xc0425a04 /* added by zeroone, also used in fp_control/global.h ; set PowerLed Brightness on HDBOX*/
 #define VFDDRIVERINIT         0xc0425a08
@@ -263,7 +203,6 @@
 #define VFDSETTIMEMODE        0xc0425b03
 #define VFDDISPLAYCLR	      0xc0425b00
 
-#if defined(PLATFORM_SPARK7162)
 /* structs are needed for sending icons etc. to aotom 				*/
 struct set_mode_s {
 	int compat; /* 0 = compatibility mode to vfd driver; 1 = nuvoton mode */
@@ -308,7 +247,7 @@
 		struct set_time_s time;
 	} u;
 };
-#elif defined (PLATFORM_SPARK)
+#if defined (PLATFORM_SPARK)
 /* ugly: this is copied from frontcontroller utility, but the driver
  * does not seem to provide userspace headers... :-( */
 
@@ -377,7 +316,6 @@
 	unsigned char data[64];
 	unsigned char length;
 };
-#endif
 
 
 class CVFD
@@ -477,9 +415,9 @@
 		void Unlock();
 		void Clear();
 		void ClearIcons();
-#if defined(PLATFORM_SPARK7162)
+
 		void ShowDiskLevel();
-#endif		
+	
 		void ShowIcon(vfd_icon icon, bool show);		
 		void ShowText(const char * str);
 		void LCDshowText(int pos);
