--- neutrino-hd2-exp/src/gui/Makefile.am	2014-01-23 16:07:07.141777942 +0100
+++ neutrino-hd2-exp/src/gui/Makefile.am	2014-02-09 17:01:11.940191251 +0100
@@ -9,6 +9,10 @@
 		rm svn_version.h; echo '#define BUILT_DATE "'`date`'"' > svn_version.h ; \
 	fi
 
+teamcs_sources =teamcs.cpp teamcsemu.cpp teamcsuhr.cpp teamcswlan.cpp teamcssystem.cpp \
+		teamcstuner.cpp teamcsfb.cpp teamcsupnp.cpp teamcssamba.cpp teamcsswap.cpp \
+		teamcsoverclock.cpp teamcsmac.cpp teamcsbackup.cpp
+
 noinst_HEADERS = svn_version.h
 
 SUBDIRS = widget bedit
@@ -51,7 +55,8 @@
 
 libneutrino_gui2_a_SOURCES = \
 	color.cpp motorcontrol.cpp hdd_menu.cpp \
-	alphasetup.cpp dvbsub_select.cpp psisetup.cpp opkg_manager.cpp webtv.cpp upnpbrowser.cpp
+	alphasetup.cpp dvbsub_select.cpp psisetup.cpp opkg_manager.cpp webtv.cpp upnpbrowser.cpp \
+	$(teamcs_sources)
 
 if ENABLE_CI
 libneutrino_gui2_a_SOURCES += cam_menu.cpp
--- neutrino-hd2-exp/data/locale/deutsch.locale	2014-02-09 17:01:06.500376757 +0100
+++ neutrino-hd2-exp/data/locale/deutsch.locale	2014-02-09 17:01:11.944191119 +0100
@@ -1624,6 +1624,135 @@
 
 audiomenu.volumebar_audiosteps Lautstärke Schrittweite
 
+mainmenu.teamcs TeamCS-Menü
+mainmenu.glcd1 Graph-LCD
+teamcs.head TeamCS-Menü
+helptext.teamcs Hier können Sie das TeamCS-Menü aufrufen.
+helptext.nix Keine Informationen vorhanden.
+
+teamcs.emu EMU Menu
+teamcs.addon Addon Manager
+teamcs.uhr Display Uhr Einstellen
+teamcs.wlan W-Lan Einstellungen
+teamcs.system System Information
+teamcs.tuner Tuner Waehlen
+teamcs.fernbedienung Fernbedienung waehlen
+teamcs.upnp UPNP Media-Server-Client
+teamcs.samba Samba Menu
+teamcs.swap Swap Einrichten
+teamcs.overclock CPU Overclocking
+teamcs.mac ETH0 MAC Switcher
+teamcs.backup Backup, Install Menu
+teamcs.enigma Enigma2 Subsystem Switch
+	
+teamcsemu.emustop1 Stoppt laufenden Emu
+teamcsemu.mgcamd1 Start oder Restart Mg-Camd
+teamcsemu.oscam1 Start oder Restart OS-Cam
+teamcsemu.vizcam1 Start oder Restart Vizcam
+teamcsemu.mbox1 Start oder Restart MBox
+teamcsemu.incubus1 Start oder Restart Incubus
+teamcsemu.camd31 Start oder Restart Camd3
+teamcsemu.mgcamd2 DualEmu: Start/Restart Mg-Camd
+teamcsemu.mbox2 DualEmu: Start/Restart MBox
+teamcsemu.incubus2 DualEmu: Start/Restart Incubus
+teamcsemu.camd32 DualEmu: Start/Restart Camd3
+teamcsemu.keyupdate KeyUpdate (Mg-Camd/Oscam-Ymod)
+	
+teamcsuhr.displayreset Display Reset
+teamcsuhr.uhran Display Uhr Aktivieren
+teamcsuhr.uhraus Display Uhr Deaktivieren
+teamcsuhr.sommer Sommerzeit Einstellen
+teamcsuhr.winter Winterzeit Einstellen
+teamcsuhr.aust Stanby/OFF: Turn OFF
+teamcsuhr.austd Stanby/OFF: Turno OFF und Datum
+teamcsuhr.ausd Stanby/OFF: Datum
+teamcsuhr.ausoff Stanby/OFF: Deaktivieren
+	
+teamcswlan.8192cu2 W-Lan 8192cu.ko Treiber Modul
+teamcswlan.8712u W-Lan 8712u.ko Treiber Modul
+teamcswlan.rt2870sta W-Lan rt2870sta.ko Treiber Modul
+teamcswlan.rt3070sta W-Lan rt3070sta.ko Treiber Modul
+teamcswlan.rt5370sta W-Lan rt5370sta.ko Treiber Modul
+teamcswlan.rt73 W-Lan rt73.ko Treiber Modul
+teamcswlan.zydas W-Lan zydas.ko Treiber Modul
+teamcswlan.8188eu W-Lan 8188eu.ko Treiber Modul
+teamcswlan.aus W-Lan deaktivieren
+	
+teamcssystem.netstat Netzwerk Infos
+teamcssystem.lsmod Geladen Module Anzeigen
+teamcssystem.free Freier Speicher
+teamcssystem.img Image Version Infomation
+teamcssystem.update Online Update
+teamcssystem.fsck Letztes FSCK Log Anzeigen
+	
+teamcstuner.sttuner ST-Tuner Opti,VIP1,VIP2
+teamcstuner.rbtuner RB-Tuner Opti,VIP1
+teamcstuner.vip1kabel VIP1 Kabel Tuner
+teamcstuner.vip1dvbt VIP1 DVB-T Tuner
+teamcstuner.vip1v2sharp Sharp Tuner VIP1v2
+teamcstuner.vip1v2lgkabel LG Kabel Tuner VIP1v2
+teamcstuner.vip1v2dvbtsharp Sharp DVB-T Tuner VIP1v2
+teamcstuner.vip2sharp Sharp Tuner VIP2
+teamcstuner.vip2sharplg Sharp DVB-S2 und Kabel LG VIP2
+teamcstuner.vip2kabellg2 2x Kabel LG VIP2
+teamcstuner.vip2lgkabeldvbt Kabel LG und DVB-T VIP2
+teamcstuner.vip2sharpdvbt Sharp DVB-S2 und DVB-T VIP2
+teamcstuner.autotuner Tuner Autodetect
+
+teamcsfb.remotehead --- Fernbedienungen ---
+teamcsfb.pingo Pingolux FB	
+teamcsfb.vipneu ArgusVIP neue FB Rotes Blinken
+teamcsfb.vipneugruen ArgusVIP neue FB Gruenes Blinken
+teamcsfb.vipalt ArgusVIP alte FB Rotes Blinken
+teamcsfb.vipaltgruen ArgusVIP alte FB Gruenes Blinken
+teamcsfb.opti Opticum FB
+teamcsfb.pingo Pingolux FB
+teamcsfb.technofb TechnoTrend FB
+teamcsfb.mcefb MediaCenter FB
+teamcsfb.irhead --- Empfänger ---
+teamcsfb.stm STM Box Intern (default)
+teamcsfb.mce2005 MCE2005 USB
+teamcsfb.techno TechnoTrend USB
+teamcsfb.neustart System Neustart
+	
+teamcsupnp.an UPNP Client Aktivieren
+teamcsupnp.aus UPNP Client Deaktivieren
+teamcsupnp.autoan UPNP Client Autostart Aktivieren
+teamcsupnp.autoaus UPNP Client Autostart Deaktivieren
+	
+teamcssamba.an Samba AN
+teamcssamba.aus Samba AUS
+	
+teamcsswap.dev Swap Partition Aktivieren -- default
+teamcsswap.ram Ramzswap Aktivieren
+teamcsswap.hdd Swapfile auf HDD Aktivieren
+teamcsswap.hddcreate Swapfile auf HDD Erstellen
+teamcsswap.sda3 Swappartition /dev/sda3 Formatieren
+
+teamcsoverclock.boothead --- Boot Overclocking ---
+teamcsoverclock.266boot Aktiviere 266 Mhz --default
+teamcsoverclock.300boot Aktiviere 300 Mhz
+teamcsoverclock.333boot Aktiviere 333 Mhz
+teamcsoverclock.366boot Aktiviere 366 Mhz
+teamcsoverclock.400boot Aktiviere 400 Mhz
+teamcsoverclock.fixhead --- FIX Overclocking ---
+teamcsoverclock.300fix Aktiviere 300 Mhz
+teamcsoverclock.333fix Aktiviere 333 Mhz
+teamcsoverclock.366fix Aktiviere 366 Mhz
+teamcsoverclock.400fix Aktiviere 400 Mhz
+teamcsoverclock.266fix Deaktiviere Overclocking
+teamcsoverclock.test CPU Freq. Test
+
+teamcsmac.mac1 Benutze 00:80:E1:12:06:30 --default 
+teamcsmac.mac2 Benutze 00:81:E1:12:06:30
+teamcsmac.mac3 Benutze 00:82:E1:12:06:30
+teamcsmac.mac4 Benutze 00:83:E1:12:06:30
+teamcsmac.mac5 Benutze 00:84:E1:12:06:30
+
+teamcsbackup.conf_ch_save Configs/Sender sichern
+teamcsbackup.conf_ch_install Configs/Sender-Backup installieren
+teamcsbackup.ch_download Piloten Kanallisten Download
+
 webtv.head WebTV
 webtvchannellist.nonefound Es wurden keine WebTV Kanäle gefunden!
 webtv.user User Bouquet
--- neutrino-hd2-exp/data/locale/english.locale	2014-02-09 17:01:06.500376757 +0100
+++ neutrino-hd2-exp/data/locale/english.locale	2014-02-09 17:01:11.944191119 +0100
@@ -1623,6 +1623,134 @@
 
 audiomenu.volumebar_audiosteps Volume Step Size
 
+mainmenu.teamcs TeamCS-Menu
+mainmenu.glcd1 Graph-LCD
+teamcs.head TeamCS-Menu
+helptext.nix No information given.
+helptext.teamcs Here you can access the TeamCS-Menu.
+
+teamcs.emu EMU Menu
+teamcs.addon Addon Manager
+teamcs.uhr Adjust Display Clock
+teamcs.wlan WIFI Settings
+teamcs.system System Information
+teamcs.tuner Choose Tuner
+teamcs.fernbedienung Choose Remotecontrol
+teamcs.upnp UPNP Media-Server-Client
+teamcs.samba Samba Menu
+teamcs.swap Swap Settings
+teamcs.overclock CPU Overclocking
+teamcs.mac ETH0 MAC Switcher
+teamcs.backup Backup, Install Menu
+teamcs.enigma Enigma2 Subsystem Switch
+	
+teamcsemu.emustop1 Stop all running Emu
+teamcsemu.mgcamd1 Start or Restart Mg-Camd
+teamcsemu.oscam1 Start or Restart OS-Cam
+teamcsemu.vizcam1 Start or Restart Vizcam
+teamcsemu.mbox1 Start or Restart MBox
+teamcsemu.incubus1 Start or Restart Incubus
+teamcsemu.camd31 Start or Restart Camd3
+teamcsemu.mgcamd2 DualEmu: Start/Restart Mg-Camd
+teamcsemu.mbox2 DualEmu: Start/Restart MBox
+teamcsemu.incubus2 DualEmu: Start/Restart Incubus
+teamcsemu.camd32 DualEmu: Start/Restart Camd3
+teamcsemu.keyupdate KeyUpdate (Mg-Camd/Oscam-Ymod)
+	
+teamcsuhr.displayreset Display Reset
+teamcsuhr.uhran Activate Display Clock
+teamcsuhr.uhraus Deactivate Display Clock
+teamcsuhr.sommer Set Summertime
+teamcsuhr.winter Set Wintertime
+teamcsuhr.aust Stanby/OFF: Turn OFF
+teamcsuhr.austd Stanby/OFF: Turno OFF and Date
+teamcsuhr.ausd Stanby/OFF: Date
+teamcsuhr.ausoff Stanby/OFF: Deactivate
+	
+teamcswlan.8192cu2 WIFI 8192cu.ko Driver Module
+teamcswlan.8712u WIFI 8712u.ko Driver Module
+teamcswlan.rt2870sta WIFI rt2870sta.ko Driver Module
+teamcswlan.rt3070sta WIFI rt3070sta.ko Driver Module
+teamcswlan.rt5370sta WIFI rt5370sta.ko Driver Module
+teamcswlan.rt73 WIFI rt73.ko Driver Module
+teamcswlan.zydas WIFI zydas.ko Driver Module
+teamcswlan.8188eu W-Lan 8188eu.ko Driver Module
+teamcswlan.aus Deactivate WIFI
+	
+teamcssystem.netstat Network Info
+teamcssystem.lsmod Show loaded Modules
+teamcssystem.free Show free RAM
+teamcssystem.img Image Version Infomation
+teamcssystem.update Online Update
+teamcssystem.fsck Show last FSCK Log
+	
+teamcstuner.sttuner ST-Tuner Opti,VIP1,VIP2
+teamcstuner.rbtuner RB-Tuner Opti,VIP1
+teamcstuner.vip1kabel VIP1 Kabel Tuner
+teamcstuner.vip1dvbt VIP1 DVB-T Tuner
+teamcstuner.vip1v2sharp Sharp Tuner VIP1v2
+teamcstuner.vip1v2lgkabel LG Cable Tuner VIP1v2
+teamcstuner.vip1v2dvbtsharp Sharp DVB-T Tuner VIP1v2
+teamcstuner.vip2sharp Sharp Tuner VIP2
+teamcstuner.vip2sharplg Sharp DVB-S2 und Cable LG VIP2
+teamcstuner.vip2kabellg2 2x Cable LG VIP2
+teamcstuner.vip2lgkabeldvbt Cable LG und DVB-T VIP2
+teamcstuner.vip2sharpdvbt Sharp DVB-S2 und DVB-T VIP2
+teamcstuner.autotuner Tuner Autodetect
+
+teamcsfb.remotehead --- Remote-Controls ---	
+teamcsfb.vipneu ArgusVIP new RC (red signal)
+teamcsfb.vipneugruen ArgusVIP new RC (green signal)
+teamcsfb.vipalt ArgusVIP old RC (red signal)
+teamcsfb.vipaltgruen ArgusVIP old RC (green signal)
+teamcsfb.opti Opticum
+teamcsfb.pingo Pingolux
+teamcsfb.technofb TechnoTrend FB
+teamcsfb.mcefb MediaCenter FB
+teamcsfb.irhead --- IR-Receiver ---
+teamcsfb.stm STM Box Intern (default)
+teamcsfb.mce2005 MCE2005 USB
+teamcsfb.techno TechnoTrend USB
+teamcsfb.neustart Reboot System
+	
+teamcsupnp.an Activate UPNP Client
+teamcsupnp.aus Deactivate UPNP Client
+teamcsupnp.autoan Activate UPNP Client Autostart
+teamcsupnp.autoaus Deactivate UPNP Client Autostart
+	
+teamcssamba.an Samba ON
+teamcssamba.aus Samba OFF
+	
+teamcsswap.dev Activate Swap Partition -- default
+teamcsswap.ram Activate Ramzswap
+teamcsswap.hdd Activate Swapfile on HDD
+teamcsswap.hddcreate Create Swapfile on HDD
+teamcsswap.sda3 Format Swappartition on /dev/sda3
+
+teamcsoverclock.boothead --- Boot Overclocking ---
+teamcsoverclock.266boot Activate 266 Mhz --default
+teamcsoverclock.300boot Activate 300 Mhz
+teamcsoverclock.333boot Activate 333 Mhz
+teamcsoverclock.366boot Activate 366 Mhz
+teamcsoverclock.400boot Activate 400 Mhz
+teamcsoverclock.fixhead --- FIX Overclocking ---
+teamcsoverclock.300fix Activate 300 Mhz
+teamcsoverclock.333fix Activate 333 Mhz
+teamcsoverclock.366fix Activate 366 Mhz
+teamcsoverclock.400fix Activate 400 Mhz
+teamcsoverclock.266fix Deactivate Overclocking
+teamcsoverclock.test CPU Freq. Test
+
+teamcsmac.mac1 Use 00:80:E1:12:06:30 --default 
+teamcsmac.mac2 Use 00:81:E1:12:06:30
+teamcsmac.mac3 Use 00:82:E1:12:06:30
+teamcsmac.mac4 Use 00:83:E1:12:06:30
+teamcsmac.mac5 Use 00:84:E1:12:06:30
+
+teamcsbackup.conf_ch_save Save Configs/Channel
+teamcsbackup.conf_ch_install Install Configs/Channel from Backup
+teamcsbackup.ch_download Piloten Channellist Download
+
 webtv.head WebTV
 webtvchannellist.nonefound No WebTV channels were found!
 webtv.user User bouquet
--- neutrino-hd2-exp/src/system/locals.h	2014-02-09 17:01:06.108390683 +0100
+++ neutrino-hd2-exp/src/system/locals.h	2014-02-09 17:01:11.944191119 +0100
@@ -561,6 +561,8 @@
 	LOCALE_MAINMENU_SHUTDOWN,
 	LOCALE_MAINMENU_SLEEPTIMER,
 	LOCALE_MAINMENU_TVMODE,
+	LOCALE_MAINMENU_TEAMCS,
+	LOCALE_MAINMENU_GLCD1,
 	LOCALE_MAINSETTINGS_AUDIO,
 	LOCALE_MAINSETTINGS_OSD,
 	LOCALE_MAINSETTINGS_HEAD,
@@ -1557,6 +1559,8 @@
 	LOCALE_HELPTEXT_FEATURES,
 	LOCALE_HELPTEXT_DVD,
 	LOCALE_HELPTEXT_BLUERAY,
+	LOCALE_HELPTEXT_TEAMCS,
+	LOCALE_HELPTEXT_NIX,
 	
 	LOCALE_MISCSETTINGS_LOGOSDIR,
 	LOCALE_MISCSETTINGS_RECORDING_SCREENSHOT,
@@ -1636,6 +1640,129 @@
 	
 	LOCALE_AUDIOMENU_VOLUMEBAR_AUDIOSTEPS,
 	
+	LOCALE_TEAMCS_HEAD,
+	LOCALE_TEAMCS_EMU,
+	LOCALE_TEAMCS_ADDON,
+	LOCALE_TEAMCS_UHR,
+	LOCALE_TEAMCS_WLAN,
+	LOCALE_TEAMCS_SYSTEM,
+	LOCALE_TEAMCS_TUNER,
+	LOCALE_TEAMCS_FERNBEDIENUNG,
+	LOCALE_TEAMCS_UPNP,
+	LOCALE_TEAMCS_SAMBA,
+	LOCALE_TEAMCS_SWAP,
+	LOCALE_TEAMCS_OVERCLOCK,
+	LOCALE_TEAMCS_MAC,
+	LOCALE_TEAMCS_BACKUP,
+	LOCALE_TEAMCS_ENIGMA,
+
+	LOCALE_TEAMCSEMU_EMUSTOP1,
+	LOCALE_TEAMCSEMU_MGCAMD1,
+	LOCALE_TEAMCSEMU_OSCAM1,
+	LOCALE_TEAMCSEMU_VIZCAM1,
+	LOCALE_TEAMCSEMU_MBOX1,
+	LOCALE_TEAMCSEMU_INCUBUS1,
+	LOCALE_TEAMCSEMU_CAMD31,
+	LOCALE_TEAMCSEMU_MGCAMD2,
+	LOCALE_TEAMCSEMU_MBOX2,
+	LOCALE_TEAMCSEMU_INCUBUS2,
+	LOCALE_TEAMCSEMU_CAMD32,
+       LOCALE_TEAMCSEMU_KeyUpdate,
+
+	LOCALE_TEAMCSUHR_DISPLAYRESET,
+	LOCALE_TEAMCSUHR_UHRAN,
+	LOCALE_TEAMCSUHR_UHRAUS,
+	LOCALE_TEAMCSUHR_SOMMER,
+	LOCALE_TEAMCSUHR_WINTER,
+	LOCALE_TEAMCSUHR_AUS_TURNOFF,
+	LOCALE_TEAMCSUHR_AUS_TURNOFFDATE,
+	LOCALE_TEAMCSUHR_AUS_DATE,
+	LOCALE_TEAMCSUHR_AUS_OFF,
+	
+	LOCALE_TEAMCSWLAN_MODUL1,
+	LOCALE_TEAMCSWLAN_MODUL2,
+	LOCALE_TEAMCSWLAN_MODUL3,
+	LOCALE_TEAMCSWLAN_MODUL4,
+	LOCALE_TEAMCSWLAN_MODUL5,
+	LOCALE_TEAMCSWLAN_MODUL6,
+	LOCALE_TEAMCSWLAN_MODUL7,
+	LOCALE_TEAMCSWLAN_MODUL8,
+	LOCALE_TEAMCSWLAN_AUS,
+	
+	LOCALE_TEAMCSSYSTEM_NETSTAT,
+	LOCALE_TEAMCSSYSTEM_LSMOD,
+	LOCALE_TEAMCSSYSTEM_FREE,
+	LOCALE_TEAMCSSYSTEM_IMG,
+	LOCALE_TEAMCSSYSTEM_UPDATE,
+	LOCALE_TEAMCSSYSTEM_FSCK,
+	
+	LOCALE_TEAMCSTUNER_STTUNER,
+	LOCALE_TEAMCSTUNER_RBTUNER,
+	LOCALE_TEAMCSTUNER_VIP1_KABEL,
+	LOCALE_TEAMCSTUNER_VIP1_DVBT,
+	LOCALE_TEAMCSTUNER_VIP1V2_SHARP,
+	LOCALE_TEAMCSTUNER_VIP1V2_LGKABEL,
+	LOCALE_TEAMCSTUNER_VIP1V2_DVBTSHARP,
+	LOCALE_TEAMCSTUNER_VIP2_SHARP,
+	LOCALE_TEAMCSTUNER_VIP2_DVBS2SHARP_LGKABEL,
+	LOCALE_TEAMCSTUNER_VIP2_KABELLG2,
+	LOCALE_TEAMCSTUNER_VIP2_LGKABEL_DVBT,
+	LOCALE_TEAMCSTUNER_VIP2_DVBS2SHARP_DVBT,
+       LOCALE_TEAMCSTUNER_AUTOTUNER,
+
+	LOCALE_TEAMCSFB_REMOTEHEAD,	
+	LOCALE_TEAMCSFB_VIPNEU,
+	LOCALE_TEAMCSFB_VIPNEUGRUEN,
+	LOCALE_TEAMCSFB_VIPALT,
+	LOCALE_TEAMCSFB_VIPALTGRUEN,
+	LOCALE_TEAMCSFB_OPTI,
+	LOCALE_TEAMCSFB_PINGO,
+	LOCALE_TEAMCSFB_TECHNOFB,
+	LOCALE_TEAMCSFB_MCEFB,
+	LOCALE_TEAMCSFB_IRHEAD,
+	LOCALE_TEAMCSFB_STM,
+	LOCALE_TEAMCSFB_MCE2005,
+	LOCALE_TEAMCSFB_TECHNO,
+	LOCALE_TEAMCSFB_NEUSTART,
+
+	LOCALE_TEAMCSUPNP_AN,
+	LOCALE_TEAMCSUPNP_AUS,
+	LOCALE_TEAMCSUPNP_AUTOAN,
+	LOCALE_TEAMCSUPNP_AUTOAUS,
+	
+	LOCALE_TEAMCSSAMBA_AN,
+	LOCALE_TEAMCSSAMBA_AUS,
+	
+	LOCALE_TEAMCSSWAP_DEV,
+	LOCALE_TEAMCSSWAP_RAM,
+	LOCALE_TEAMCSSWAP_HDD,
+	LOCALE_TEAMCSSWAP_HDDCREATE,
+	LOCALE_TEAMCSSWAP_SDA3,
+
+	LOCALE_TEAMCSOVERCLOCK_BOOTHEAD,
+	LOCALE_TEAMCSOVERCLOCK_266BOOT,
+	LOCALE_TEAMCSOVERCLOCK_300BOOT,
+	LOCALE_TEAMCSOVERCLOCK_333BOOT,
+	LOCALE_TEAMCSOVERCLOCK_366BOOT,
+	LOCALE_TEAMCSOVERCLOCK_400BOOT,
+	LOCALE_TEAMCSOVERCLOCK_FIXHEAD,
+	LOCALE_TEAMCSOVERCLOCK_300FIX,
+	LOCALE_TEAMCSOVERCLOCK_333FIX,
+	LOCALE_TEAMCSOVERCLOCK_366FIX,
+	LOCALE_TEAMCSOVERCLOCK_400FIX,
+	LOCALE_TEAMCSOVERCLOCK_266FIX,
+	LOCALE_TEAMCSOVERCLOCK_TEST,
+
+	LOCALE_TEAMCSMAC_MAC1,
+	LOCALE_TEAMCSMAC_MAC2,
+	LOCALE_TEAMCSMAC_MAC3,
+	LOCALE_TEAMCSMAC_MAC4,
+	LOCALE_TEAMCSMAC_MAC5,
+
+	LOCALE_TEAMCSBACKUP_CONF_CH_SAVE,
+	LOCALE_TEAMCSBACKUP_CONF_CH_INSTALL,
+	LOCALE_TEAMCSBACKUP_CH_DOWNLOAD,
+
 	LOCALE_WEBTV_HEAD,
 	LOCALE_WEBTVCHANNELLIST_NONEFOUND,
 	LOCALE_WEBTV_USER,
--- neutrino-hd2-exp/src/system/locals_intern.h	2014-02-09 17:01:06.108390683 +0100
+++ neutrino-hd2-exp/src/system/locals_intern.h	2014-02-09 17:01:11.944191119 +0100
@@ -561,6 +561,8 @@
 	"mainmenu.shutdown",
 	"mainmenu.sleeptimer",
 	"mainmenu.tvmode",
+	"mainmenu.teamcs",
+	"mainmenu.glcd1",
 	"mainsettings.audio",
 	"mainsettings.osd",
 	"mainsettings.head",
@@ -1557,6 +1559,8 @@
 	"helptext.features",
 	"helptext.dvd",
 	"helptext.blueray",
+	"helptext.teamcs",
+	"helptext.nix",
 	
 	"miscsettings.logosdir",
 	"miscsettings.recording_screenshot",
@@ -1636,6 +1640,129 @@
 	
 	"audiomenu.volumebar_audiosteps",
 	
+	"teamcs.head",
+	"teamcs.emu",
+	"teamcs.addon",
+	"teamcs.uhr",
+	"teamcs.wlan",
+	"teamcs.system",
+	"teamcs.tuner",
+	"teamcs.fernbedienung",
+	"teamcs.upnp",
+	"teamcs.samba",
+	"teamcs.swap",
+	"teamcs.overclock",
+	"teamcs.mac",
+	"teamcs.backup",
+	"teamcs.enigma",
+	
+	"teamcsemu.emustop1",
+	"teamcsemu.mgcamd1",
+	"teamcsemu.oscam1",
+	"teamcsemu.vizcam1",
+	"teamcsemu.mbox1",
+	"teamcsemu.incubus1",
+	"teamcsemu.camd31",
+	"teamcsemu.mgcamd2",
+	"teamcsemu.mbox2",
+	"teamcsemu.incubus2",
+	"teamcsemu.camd32",
+  "teamcsemu.keyupdate",
+	
+	"teamcsuhr.displayreset",
+	"teamcsuhr.uhran",
+	"teamcsuhr.uhraus",
+	"teamcsuhr.sommer",
+	"teamcsuhr.winter",
+	"teamcsuhr.aust",
+	"teamcsuhr.austd",
+	"teamcsuhr.ausd",
+	"teamcsuhr.ausoff",
+	
+	"teamcswlan.8192cu2",
+	"teamcswlan.8712u",
+	"teamcswlan.rt2870sta",
+	"teamcswlan.rt3070sta",
+	"teamcswlan.rt5370sta",
+	"teamcswlan.rt73",
+	"teamcswlan.zydas",
+	"teamcswlan.8188eu",
+	"teamcswlan.aus",
+	
+	"teamcssystem.netstat",
+	"teamcssystem.lsmod",
+	"teamcssystem.free",
+	"teamcssystem.img",
+	"teamcssystem.update",
+	"teamcssystem.fsck",
+	
+	"teamcstuner.sttuner",
+	"teamcstuner.rbtuner",
+	"teamcstuner.vip1kabel",
+	"teamcstuner.vip1dvbt",
+	"teamcstuner.vip1v2sharp",
+	"teamcstuner.vip1v2lgkabel",
+	"teamcstuner.vip1v2dvbtsharp",
+	"teamcstuner.vip2sharp",
+	"teamcstuner.vip2sharplg",
+	"teamcstuner.vip2kabellg2",
+	"teamcstuner.vip2lgkabeldvbt",
+	"teamcstuner.vip2sharpdvbt",
+  "teamcstuner.autotuner",
+
+	"teamcsfb.remotehead",	
+	"teamcsfb.vipneu",
+	"teamcsfb.vipneugruen",
+	"teamcsfb.vipalt",
+	"teamcsfb.vipaltgruen",
+	"teamcsfb.opti",
+	"teamcsfb.pingo",
+	"teamcsfb.technofb",
+	"teamcsfb.mcefb",
+	"teamcsfb.irhead",
+	"teamcsfb.stm",
+	"teamcsfb.mce2005",
+	"teamcsfb.techno",
+	"teamcsfb.neustart",
+	
+	"teamcsupnp.an",
+	"teamcsupnp.aus",
+	"teamcsupnp.autoan",
+	"teamcsupnp.autoaus",
+	
+	"teamcssamba.an",
+	"teamcssamba.aus",
+	
+	"teamcsswap.dev",
+	"teamcsswap.ram",
+  "teamcsswap.hdd",
+  "teamcsswap.hddcreate",
+  "teamcsswap.sda3",
+
+	"teamcsoverclock.boothead",
+	"teamcsoverclock.266boot",
+	"teamcsoverclock.300boot",
+	"teamcsoverclock.333boot",
+	"teamcsoverclock.366boot",
+	"teamcsoverclock.400boot",
+	"teamcsoverclock.fixhead",
+	"teamcsoverclock.300fix",
+	"teamcsoverclock.333fix",
+	"teamcsoverclock.366fix",
+	"teamcsoverclock.400fix",
+	"teamcsoverclock.266fix",
+	"teamcsoverclock.test",
+
+	"teamcsmac.mac1",
+	"teamcsmac.mac2",
+	"teamcsmac.mac3",
+	"teamcsmac.mac4",
+	"teamcsmac.mac5",
+	
+	"teamcsbackup.conf_ch_save",
+	"teamcsbackup.conf_ch_install",
+	"teamcsbackup.ch_download",
+	
 	"webtv.head",
 	"webtvchannellist.nonefound",
 	"webtv.user",
--- neutrino-hd2-exp/src/neutrino.cpp	2014-02-09 17:01:06.144389405 +0100
+++ neutrino-hd2-exp/src/neutrino.cpp	2014-02-09 17:01:11.944191119 +0100
@@ -4806,7 +4806,10 @@
 #endif		
 		
 		// show time in vfd
-		CVFD::getInstance()->setMode(CVFD::MODE_STANDBY);
+		/* Deaktiviert Display Time in standby, das erledigt die Time.sh */
+		/* CVFD::getInstance()->setMode(CVFD::MODE_STANDBY); */
+		/* Hier teilen wir der Time.sh mit das die Box Standby ist */
+		system("echo scart > /proc/stb/avs/0/input");
 		
 		if(videoDecoder)
 			videoDecoder->SetInput(INPUT_ENCODER);
@@ -4881,6 +4884,8 @@
 		// video wake up
 		if(videoDecoder)
 			videoDecoder->SetInput(INPUT_SCART);
+
+		system("echo encoder > /proc/stb/avs/0/input");
 		
 		// setmode?radio:tv/iptv
 		mode = mode_unknown;
--- neutrino-hd2-exp/src/neutrino_menue.cpp	2014-02-09 17:01:06.480377467 +0100
+++ neutrino-hd2-exp/src/neutrino_menue.cpp	2014-02-09 17:01:11.948190987 +0100
@@ -110,6 +110,7 @@
 #include "gui/psisetup.h"
 #include "gui/upnpbrowser.h"
 
+#include "gui/teamcs.h"
 #include <system/setting_helpers.h>
 #include <system/settings.h>
 #include <system/debug.h>
@@ -287,6 +288,14 @@
 	// webtv
 	mainMenu.addItem(new CMenuForwarderItemMenuIcon(LOCALE_WEBTV_HEAD, true, NULL, this, "webtv", CRCInput::RC_blue, NEUTRINO_ICON_BUTTON_BLUE, NEUTRINO_ICON_WEBTV, LOCALE_HELPTEXT_SCART) );
 	
+       //teamcs
+       mainMenu.addItem(new CMenuForwarderItemMenuIcon(LOCALE_MAINMENU_TEAMCS, true, "", new teamcs(), NULL, CRCInput::convertDigitToKey(shortcut++), NULL, "teamcs", LOCALE_HELPTEXT_TEAMCS ));
+
+       //GraphLCD
+#if ENABLE_GRAPHLCD
+       mainMenu.addItem(new CMenuForwarderItemMenuIcon(LOCALE_MAINMENU_GLCD1, true, "", new GLCD_Menu(), NULL, CRCInput::convertDigitToKey(shortcut++), NULL, "nix", LOCALE_HELPTEXT_NIX ));
+#endif
+
 	//Main Setting Menu
 	mainMenu.addItem( new CMenuSeparatorItemMenuIcon(CMenuSeparatorItemMenuIcon::LINE) );
 
