--- neutrino-hd2-exp.org/data/locale/deutsch.locale	2013-06-13 18:25:06.000000000 +0200
+++ neutrino-hd2-exp/data/locale/deutsch.locale	2013-06-18 18:52:11.000000000 +0200
@@ -1593,6 +1593,119 @@ keybinding.savekeymap_hint keymap wird g
 
 audiomenu.volumebar_audiosteps Lautstärke Schrittweite
 
+mainmenu.teamcs TeamCS-Menü
+mainmenu.glcd1 Graph-LCD
+teamcs.head TeamCS-Menü
+helptext.teamcs Hier können Sie das TeamCS-Menü aufrufen.
+helptext.nix Keine Informationen vorhanden.
+
+teamcs.emu EMU Menu
+teamcs.addon Addon Manager
+teamcs.uhr Display Uhr Einstellen
+teamcs.wlan W-Lan Einstellungen
+teamcs.system System Information
+teamcs.tuner Tuner Waehlen
+teamcs.fernbedienung Fernbedienung waehlen
+teamcs.upnp UPNP Media-Server-Client
+teamcs.samba Samba Menu
+teamcs.swap Swap Einrichten
+teamcs.overclock CPU Overclocking
+teamcs.mac ETH0 MAC Switcher
+teamcs.enigma Enigma2 Subsystem Switch
+	
+teamcsemu.emustop1 Stoppt laufenden Emu
+teamcsemu.mgcamd1 Start oder Restart Mg-Camd
+teamcsemu.oscam1 Start oder Restart OS-Cam
+teamcsemu.vizcam1 Start oder Restart Vizcam
+teamcsemu.mbox1 Start oder Restart MBox
+teamcsemu.incubus1 Start oder Restart Incubus
+teamcsemu.camd31 Start oder Restart Camd3
+teamcsemu.mgcamd2 DualEmu: Start/Restart Mg-Camd
+teamcsemu.mbox2 DualEmu: Start/Restart MBox
+teamcsemu.incubus2 DualEmu: Start/Restart Incubus
+teamcsemu.camd32 DualEmu: Start/Restart Camd3
+	
+teamcsuhr.displayreset Display Reset
+teamcsuhr.uhran Display Uhr Aktivieren
+teamcsuhr.uhraus Display Uhr Deaktivieren
+teamcsuhr.sommer Sommerzeit Einstellen
+teamcsuhr.winter Winterzeit Einstellen
+teamcsuhr.aust Stanby/OFF: Turn OFF
+teamcsuhr.austd Stanby/OFF: Turno OFF und Datum
+teamcsuhr.ausd Stanby/OFF: Datum
+teamcsuhr.ausoff Stanby/OFF: Deaktivieren
+	
+teamcswlan.8192cu2 W-Lan 8192cu.ko Treiber Modul
+teamcswlan.8712u W-Lan 8712u.ko Treiber Modul
+teamcswlan.rt2870sta W-Lan rt2870sta.ko Treiber Modul
+teamcswlan.rt3070sta W-Lan rt3070sta.ko Treiber Modul
+teamcswlan.rt5370sta W-Lan rt5370sta.ko Treiber Modul
+teamcswlan.rt73 W-Lan rt73.ko Treiber Modul
+teamcswlan.zydas W-Lan zydas.ko Treiber Modul
+teamcswlan.aus W-Lan deaktivieren
+	
+teamcssystem.netstat Netzwerk Infos
+teamcssystem.lsmod Geladen Module Anzeigen
+teamcssystem.free Freier Speicher
+teamcssystem.img Image Version Infomation
+teamcssystem.update Online Update
+teamcssystem.fsck Letztes FSCK Log Anzeigen
+	
+teamcstuner.sttuner ST-Tuner Opti,VIP1,VIP2
+teamcstuner.rbtuner RB-Tuner Opti,VIP1
+teamcstuner.vip1kabel VIP1 Kabel Tuner
+teamcstuner.vip1dvbt VIP1 DVB-T Tuner
+teamcstuner.vip1v2sharp Sharp Tuner VIP1v2
+teamcstuner.vip1v2lgkabel LG Kabel Tuner VIP1v2
+teamcstuner.vip1v2dvbtsharp Sharp DVB-T Tuner VIP1v2
+teamcstuner.vip2sharp Sharp Tuner VIP2
+teamcstuner.vip2sharplg Sharp DVB-S2 und Kabel LG VIP2
+teamcstuner.vip2kabellg2 2x Kabel LG VIP2
+teamcstuner.vip2lgkabeldvbt Kabel LG und DVB-T VIP2
+teamcstuner.vip2sharpdvbt Sharp DVB-S2 und DVB-T VIP2
+	
+teamcsfb.vipneu ArgusVIP neue FB Rotes Blinken
+teamcsfb.vipneugruen ArgusVIP neue FB Gruenes Blinken
+teamcsfb.vipalt ArgusVIP alte FB Rotes Blinken
+teamcsfb.vipaltgruen ArgusVIP alte FB Gruenes Blinken
+teamcsfb.opti Opticum FB
+teamcsfb.pingo Pingolux FB
+teamcsfb.neustart System Neustart
+	
+teamcsupnp.an UPNP Client Aktivieren
+teamcsupnp.aus UPNP Client Deaktivieren
+teamcsupnp.autoan UPNP Client Autostart Aktivieren
+teamcsupnp.autoaus UPNP Client Autostart Deaktivieren
+	
+teamcssamba.an Samba AN
+teamcssamba.aus Samba AUS
+	
+teamcsswap.dev Swap Partition Aktivieren -- default
+teamcsswap.ram Ramzswap Aktivieren
+teamcsswap.hdd Swapfile auf HDD Aktivieren
+teamcsswap.hddcreate Swapfile auf HDD Erstellen
+teamcsswap.sda3 Swappartition /dev/sda3 Formatieren
+
+teamcsoverclock.boothead --- Boot Overclocking ---
+teamcsoverclock.266boot Aktiviere 266 Mhz --default
+teamcsoverclock.300boot Aktiviere 300 Mhz
+teamcsoverclock.333boot Aktiviere 333 Mhz
+teamcsoverclock.366boot Aktiviere 366 Mhz
+teamcsoverclock.400boot Aktiviere 400 Mhz
+teamcsoverclock.fixhead --- FIX Overclocking ---
+teamcsoverclock.300fix Aktiviere 300 Mhz
+teamcsoverclock.333fix Aktiviere 333 Mhz
+teamcsoverclock.366fix Aktiviere 366 Mhz
+teamcsoverclock.400fix Aktiviere 400 Mhz
+teamcsoverclock.266fix Deaktiviere Overclocking
+teamcsoverclock.test CPU Freq. Test
+
+teamcsmac.mac1 Benutze 00:80:E1:12:06:30 --default 
+teamcsmac.mac2 Benutze 00:81:E1:12:06:30
+teamcsmac.mac3 Benutze 00:82:E1:12:06:30
+teamcsmac.mac4 Benutze 00:83:E1:12:06:30
+teamcsmac.mac5 Benutze 00:84:E1:12:06:30
+
 webtv.head WebTV
 
 menu.hint_ytplay Wiedergabe von ausgewählten Youtube Feeds
--- neutrino-hd2-exp.org/data/locale/english.locale	2013-06-13 18:25:06.000000000 +0200
+++ neutrino-hd2-exp/data/locale/english.locale	2013-06-18 18:52:11.000000000 +0200
@@ -1592,6 +1592,119 @@ keybinding.savekeymap_hint Saving keymap
 
 audiomenu.volumebar_audiosteps Volume Step Size
 
+mainmenu.teamcs TeamCS-Menu
+mainmenu.glcd1 Graph-LCD
+teamcs.head TeamCS-Menu
+helptext.nix No information given.
+helptext.teamcs Here you can access the TeamCS-Menu.
+
+teamcs.emu EMU Menu
+teamcs.addon Addon Manager
+teamcs.uhr Adjust Display Clock
+teamcs.wlan WIFI Settings
+teamcs.system System Information
+teamcs.tuner Choose Tuner
+teamcs.fernbedienung Choose Remotecontrol
+teamcs.upnp UPNP Media-Server-Client
+teamcs.samba Samba Menu
+teamcs.swap Swap Settings
+teamcs.overclock CPU Overclocking
+teamcs.mac ETH0 MAC Switcher
+teamcs.enigma Enigma2 Subsystem Switch
+	
+teamcsemu.emustop1 Stop all running Emu
+teamcsemu.mgcamd1 Start or Restart Mg-Camd
+teamcsemu.oscam1 Start or Restart OS-Cam
+teamcsemu.vizcam1 Start or Restart Vizcam
+teamcsemu.mbox1 Start or Restart MBox
+teamcsemu.incubus1 Start or Restart Incubus
+teamcsemu.camd31 Start or Restart Camd3
+teamcsemu.mgcamd2 DualEmu: Start/Restart Mg-Camd
+teamcsemu.mbox2 DualEmu: Start/Restart MBox
+teamcsemu.incubus2 DualEmu: Start/Restart Incubus
+teamcsemu.camd32 DualEmu: Start/Restart Camd3
+	
+teamcsuhr.displayreset Display Reset
+teamcsuhr.uhran Activate Display Clock
+teamcsuhr.uhraus Deactivate Display Clock
+teamcsuhr.sommer Set Summertime
+teamcsuhr.winter Set Wintertime
+teamcsuhr.aust Stanby/OFF: Turn OFF
+teamcsuhr.austd Stanby/OFF: Turno OFF and Date
+teamcsuhr.ausd Stanby/OFF: Date
+teamcsuhr.ausoff Stanby/OFF: Deactivate
+	
+teamcswlan.8192cu2 WIFI 8192cu.ko Driver Module
+teamcswlan.8712u WIFI 8712u.ko Driver Module
+teamcswlan.rt2870sta WIFI rt2870sta.ko Driver Module
+teamcswlan.rt3070sta WIFI rt3070sta.ko Driver Module
+teamcswlan.rt5370sta WIFI rt5370sta.ko Driver Module
+teamcswlan.rt73 WIFI rt73.ko Driver Module
+teamcswlan.zydas WIFI zydas.ko Driver Module
+teamcswlan.aus Deactivate WIFI
+	
+teamcssystem.netstat Network Info
+teamcssystem.lsmod Show loaded Modules
+teamcssystem.free Show free RAM
+teamcssystem.img Image Version Infomation
+teamcssystem.update Online Update
+teamcssystem.fsck Show last FSCK Log
+	
+teamcstuner.sttuner ST-Tuner Opti,VIP1,VIP2
+teamcstuner.rbtuner RB-Tuner Opti,VIP1
+teamcstuner.vip1kabel VIP1 Kabel Tuner
+teamcstuner.vip1dvbt VIP1 DVB-T Tuner
+teamcstuner.vip1v2sharp Sharp Tuner VIP1v2
+teamcstuner.vip1v2lgkabel LG Cable Tuner VIP1v2
+teamcstuner.vip1v2dvbtsharp Sharp DVB-T Tuner VIP1v2
+teamcstuner.vip2sharp Sharp Tuner VIP2
+teamcstuner.vip2sharplg Sharp DVB-S2 und Cable LG VIP2
+teamcstuner.vip2kabellg2 2x Cable LG VIP2
+teamcstuner.vip2lgkabeldvbt Cable LG und DVB-T VIP2
+teamcstuner.vip2sharpdvbt Sharp DVB-S2 und DVB-T VIP2
+	
+teamcsfb.vipneu ArgusVIP new RC (red signal)
+teamcsfb.vipneugruen ArgusVIP new RC (green signal)
+teamcsfb.vipalt ArgusVIP old RC (red signal)
+teamcsfb.vipaltgruen ArgusVIP old RC (green signal)
+teamcsfb.opti Opticum
+teamcsfb.pingo Pingolux
+teamcsfb.neustart Reboot System
+	
+teamcsupnp.an Activate UPNP Client
+teamcsupnp.aus Deactivate UPNP Client
+teamcsupnp.autoan Activate UPNP Client Autostart
+teamcsupnp.autoaus Deactivate UPNP Client Autostart
+	
+teamcssamba.an Samba ON
+teamcssamba.aus Samba OFF
+	
+teamcsswap.dev Activate Swap Partition -- default
+teamcsswap.ram Activate Ramzswap
+teamcsswap.hdd Activate Swapfile on HDD
+teamcsswap.hddcreate Create Swapfile on HDD
+teamcsswap.sda3 Format Swappartition on /dev/sda3
+
+teamcsoverclock.boothead --- Boot Overclocking ---
+teamcsoverclock.266boot Activate 266 Mhz --default
+teamcsoverclock.300boot Activate 300 Mhz
+teamcsoverclock.333boot Activate 333 Mhz
+teamcsoverclock.366boot Activate 366 Mhz
+teamcsoverclock.400boot Activate 400 Mhz
+teamcsoverclock.fixhead --- FIX Overclocking ---
+teamcsoverclock.300fix Activate 300 Mhz
+teamcsoverclock.333fix Activate 333 Mhz
+teamcsoverclock.366fix Activate 366 Mhz
+teamcsoverclock.400fix Activate 400 Mhz
+teamcsoverclock.266fix Deactivate Overclocking
+teamcsoverclock.test CPU Freq. Test
+
+teamcsmac.mac1 Use 00:80:E1:12:06:30 --default 
+teamcsmac.mac2 Use 00:81:E1:12:06:30
+teamcsmac.mac3 Use 00:82:E1:12:06:30
+teamcsmac.mac4 Use 00:83:E1:12:06:30
+teamcsmac.mac5 Use 00:84:E1:12:06:30
+
 webtv.head WebTV
 
 menu.hint_ytplay Play selected youtube feeds
--- neutrino-hd2-exp.org/src/system/locals.h	2013-06-13 18:25:05.000000000 +0200
+++ neutrino-hd2-exp/src/system/locals.h	2013-06-18 18:52:11.000000000 +0200
@@ -556,6 +556,8 @@ typedef enum {
 	LOCALE_MAINMENU_SHUTDOWN,
 	LOCALE_MAINMENU_SLEEPTIMER,
 	LOCALE_MAINMENU_TVMODE,
+	LOCALE_MAINMENU_TEAMCS,
+	LOCALE_MAINMENU_GLCD1,
 	LOCALE_MAINSETTINGS_AUDIO,
 	LOCALE_MAINSETTINGS_OSD,
 	LOCALE_MAINSETTINGS_HEAD,
@@ -1544,6 +1546,8 @@ typedef enum {
 	LOCALE_HELPTEXT_FEATURES,
 	LOCALE_HELPTEXT_DVD,
 	LOCALE_HELPTEXT_BLUERAY,
+	LOCALE_HELPTEXT_TEAMCS,
+	LOCALE_HELPTEXT_NIX,
 	
 	LOCALE_MISCSETTINGS_LOGOSDIR,
 	
@@ -1606,6 +1610,114 @@ typedef enum {
 	
 	LOCALE_AUDIOMENU_VOLUMEBAR_AUDIOSTEPS,
 	
+	LOCALE_TEAMCS_HEAD,
+	LOCALE_TEAMCS_EMU,
+	LOCALE_TEAMCS_ADDON,
+	LOCALE_TEAMCS_UHR,
+	LOCALE_TEAMCS_WLAN,
+	LOCALE_TEAMCS_SYSTEM,
+	LOCALE_TEAMCS_TUNER,
+	LOCALE_TEAMCS_FERNBEDIENUNG,
+	LOCALE_TEAMCS_UPNP,
+	LOCALE_TEAMCS_SAMBA,
+	LOCALE_TEAMCS_SWAP,
+	LOCALE_TEAMCS_OVERCLOCK,
+	LOCALE_TEAMCS_MAC,
+	LOCALE_TEAMCS_ENIGMA,
+
+	LOCALE_TEAMCSEMU_EMUSTOP1,
+	LOCALE_TEAMCSEMU_MGCAMD1,
+	LOCALE_TEAMCSEMU_OSCAM1,
+	LOCALE_TEAMCSEMU_VIZCAM1,
+	LOCALE_TEAMCSEMU_MBOX1,
+	LOCALE_TEAMCSEMU_INCUBUS1,
+	LOCALE_TEAMCSEMU_CAMD31,
+	LOCALE_TEAMCSEMU_MGCAMD2,
+	LOCALE_TEAMCSEMU_MBOX2,
+	LOCALE_TEAMCSEMU_INCUBUS2,
+	LOCALE_TEAMCSEMU_CAMD32,
+
+	LOCALE_TEAMCSUHR_DISPLAYRESET,
+	LOCALE_TEAMCSUHR_UHRAN,
+	LOCALE_TEAMCSUHR_UHRAUS,
+	LOCALE_TEAMCSUHR_SOMMER,
+	LOCALE_TEAMCSUHR_WINTER,
+	LOCALE_TEAMCSUHR_AUS_TURNOFF,
+	LOCALE_TEAMCSUHR_AUS_TURNOFFDATE,
+	LOCALE_TEAMCSUHR_AUS_DATE,
+	LOCALE_TEAMCSUHR_AUS_OFF,
+	
+	LOCALE_TEAMCSWLAN_MODUL1,
+	LOCALE_TEAMCSWLAN_MODUL2,
+	LOCALE_TEAMCSWLAN_MODUL3,
+	LOCALE_TEAMCSWLAN_MODUL4,
+	LOCALE_TEAMCSWLAN_MODUL5,
+	LOCALE_TEAMCSWLAN_MODUL6,
+	LOCALE_TEAMCSWLAN_MODUL7,
+	LOCALE_TEAMCSWLAN_AUS,
+	
+	LOCALE_TEAMCSSYSTEM_NETSTAT,
+	LOCALE_TEAMCSSYSTEM_LSMOD,
+	LOCALE_TEAMCSSYSTEM_FREE,
+	LOCALE_TEAMCSSYSTEM_IMG,
+	LOCALE_TEAMCSSYSTEM_UPDATE,
+	LOCALE_TEAMCSSYSTEM_FSCK,
+	
+	LOCALE_TEAMCSTUNER_STTUNER,
+	LOCALE_TEAMCSTUNER_RBTUNER,
+	LOCALE_TEAMCSTUNER_VIP1_KABEL,
+	LOCALE_TEAMCSTUNER_VIP1_DVBT,
+	LOCALE_TEAMCSTUNER_VIP1V2_SHARP,
+	LOCALE_TEAMCSTUNER_VIP1V2_LGKABEL,
+	LOCALE_TEAMCSTUNER_VIP1V2_DVBTSHARP,
+	LOCALE_TEAMCSTUNER_VIP2_SHARP,
+	LOCALE_TEAMCSTUNER_VIP2_DVBS2SHARP_LGKABEL,
+	LOCALE_TEAMCSTUNER_VIP2_KABELLG2,
+	LOCALE_TEAMCSTUNER_VIP2_LGKABEL_DVBT,
+	LOCALE_TEAMCSTUNER_VIP2_DVBS2SHARP_DVBT,
+	
+	LOCALE_TEAMCSFB_VIPNEU,
+	LOCALE_TEAMCSFB_VIPNEUGRUEN,
+	LOCALE_TEAMCSFB_VIPALT,
+	LOCALE_TEAMCSFB_VIPALTGRUEN,
+	LOCALE_TEAMCSFB_OPTI,
+	LOCALE_TEAMCSFB_PINGO,
+	LOCALE_TEAMCSFB_NEUSTART,
+
+	LOCALE_TEAMCSUPNP_AN,
+	LOCALE_TEAMCSUPNP_AUS,
+	LOCALE_TEAMCSUPNP_AUTOAN,
+	LOCALE_TEAMCSUPNP_AUTOAUS,
+	
+	LOCALE_TEAMCSSAMBA_AN,
+	LOCALE_TEAMCSSAMBA_AUS,
+	
+	LOCALE_TEAMCSSWAP_DEV,
+	LOCALE_TEAMCSSWAP_RAM,
+	LOCALE_TEAMCSSWAP_HDD,
+	LOCALE_TEAMCSSWAP_HDDCREATE,
+	LOCALE_TEAMCSSWAP_SDA3,
+
+	LOCALE_TEAMCSOVERCLOCK_BOOTHEAD,
+	LOCALE_TEAMCSOVERCLOCK_266BOOT,
+	LOCALE_TEAMCSOVERCLOCK_300BOOT,
+	LOCALE_TEAMCSOVERCLOCK_333BOOT,
+	LOCALE_TEAMCSOVERCLOCK_366BOOT,
+	LOCALE_TEAMCSOVERCLOCK_400BOOT,
+	LOCALE_TEAMCSOVERCLOCK_FIXHEAD,
+	LOCALE_TEAMCSOVERCLOCK_300FIX,
+	LOCALE_TEAMCSOVERCLOCK_333FIX,
+	LOCALE_TEAMCSOVERCLOCK_366FIX,
+	LOCALE_TEAMCSOVERCLOCK_400FIX,
+	LOCALE_TEAMCSOVERCLOCK_266FIX,
+	LOCALE_TEAMCSOVERCLOCK_TEST,
+
+	LOCALE_TEAMCSMAC_MAC1,
+	LOCALE_TEAMCSMAC_MAC2,
+	LOCALE_TEAMCSMAC_MAC3,
+	LOCALE_TEAMCSMAC_MAC4,
+	LOCALE_TEAMCSMAC_MAC5,
+
 	LOCALE_WEBTV_HEAD,
 	
 	//youtube
--- neutrino-hd2-exp.org/src/system/locals_intern.h	2013-06-13 18:25:05.000000000 +0200
+++ neutrino-hd2-exp/src/system/locals_intern.h	2013-06-18 18:52:11.000000000 +0200
@@ -556,6 +556,8 @@ const char * locale_real_names[] = {
 	"mainmenu.shutdown",
 	"mainmenu.sleeptimer",
 	"mainmenu.tvmode",
+	"mainmenu.teamcs",
+	"mainmenu.glcd1",
 	"mainsettings.audio",
 	"mainsettings.osd",
 	"mainsettings.head",
@@ -1544,6 +1546,8 @@ const char * locale_real_names[] = {
 	"helptext.features",
 	"helptext.dvd",
 	"helptext.blueray",
+	"helptext.teamcs",
+	"helptext.nix",
 	
 	"miscsettings.logosdir",
 	
@@ -1606,6 +1610,114 @@ const char * locale_real_names[] = {
 	
 	"audiomenu.volumebar_audiosteps",
 	
+	"teamcs.head",
+	"teamcs.emu",
+	"teamcs.addon",
+	"teamcs.uhr",
+	"teamcs.wlan",
+	"teamcs.system",
+	"teamcs.tuner",
+	"teamcs.fernbedienung",
+	"teamcs.upnp",
+	"teamcs.samba",
+	"teamcs.swap",
+	"teamcs.overclock",
+	"teamcs.mac",
+	"teamcs.enigma",
+	
+	"teamcsemu.emustop1",
+	"teamcsemu.mgcamd1",
+	"teamcsemu.oscam1",
+	"teamcsemu.vizcam1",
+	"teamcsemu.mbox1",
+	"teamcsemu.incubus1",
+	"teamcsemu.camd31",
+	"teamcsemu.mgcamd2",
+	"teamcsemu.mbox2",
+	"teamcsemu.incubus2",
+	"teamcsemu.camd32",
+	
+	"teamcsuhr.displayreset",
+	"teamcsuhr.uhran",
+	"teamcsuhr.uhraus",
+	"teamcsuhr.sommer",
+	"teamcsuhr.winter",
+	"teamcsuhr.aust",
+	"teamcsuhr.austd",
+	"teamcsuhr.ausd",
+	"teamcsuhr.ausoff",
+	
+	"teamcswlan.8192cu2",
+	"teamcswlan.8712u",
+	"teamcswlan.rt2870sta",
+	"teamcswlan.rt3070sta",
+	"teamcswlan.rt5370sta",
+	"teamcswlan.rt73",
+	"teamcswlan.zydas",
+	"teamcswlan.aus",
+	
+	"teamcssystem.netstat",
+	"teamcssystem.lsmod",
+	"teamcssystem.free",
+	"teamcssystem.img",
+	"teamcssystem.update",
+	"teamcssystem.fsck",
+	
+	"teamcstuner.sttuner",
+	"teamcstuner.rbtuner",
+	"teamcstuner.vip1kabel",
+	"teamcstuner.vip1dvbt",
+	"teamcstuner.vip1v2sharp",
+	"teamcstuner.vip1v2lgkabel",
+	"teamcstuner.vip1v2dvbtsharp",
+	"teamcstuner.vip2sharp",
+	"teamcstuner.vip2sharplg",
+	"teamcstuner.vip2kabellg2",
+	"teamcstuner.vip2lgkabeldvbt",
+	"teamcstuner.vip2sharpdvbt",
+	
+	"teamcsfb.vipneu",
+	"teamcsfb.vipneugruen",
+	"teamcsfb.vipalt",
+	"teamcsfb.vipaltgruen",
+	"teamcsfb.opti",
+	"teamcsfb.pingo",
+	"teamcsfb.neustart",
+	
+	"teamcsupnp.an",
+	"teamcsupnp.aus",
+	"teamcsupnp.autoan",
+	"teamcsupnp.autoaus",
+	
+	"teamcssamba.an",
+	"teamcssamba.aus",
+	
+	"teamcsswap.dev",
+	"teamcsswap.ram",
+       "teamcsswap.hdd",
+       "teamcsswap.hddcreate",
+       "teamcsswap.sda3",
+
+	"teamcsoverclock.boothead",
+	"teamcsoverclock.266boot",
+	"teamcsoverclock.300boot",
+	"teamcsoverclock.333boot",
+	"teamcsoverclock.366boot",
+	"teamcsoverclock.400boot",
+	"teamcsoverclock.fixhead",
+	"teamcsoverclock.300fix",
+	"teamcsoverclock.333fix",
+	"teamcsoverclock.366fix",
+	"teamcsoverclock.400fix",
+	"teamcsoverclock.266fix",
+	"teamcsoverclock.test",
+
+	"teamcsmac.mac1",
+	"teamcsmac.mac2",
+	"teamcsmac.mac3",
+	"teamcsmac.mac4",
+	"teamcsmac.mac5",
+	
 	"webtv.head",
 	
 	// youtube
--- neutrino-hd2-exp.org/src/neutrino.cpp	2013-06-13 18:25:05.000000000 +0200
+++ neutrino-hd2-exp/src/neutrino.cpp	2013-06-18 18:52:11.000000000 +0200
@@ -4373,7 +4373,10 @@ void CNeutrinoApp::standbyMode( bool bOn
 #endif		
 		
 		// show time in vfd
-		CVFD::getInstance()->setMode(CVFD::MODE_STANDBY);
+		/* Deaktiviert Display Time in standby, das erledigt die Time.sh */
+		/* CVFD::getInstance()->setMode(CVFD::MODE_STANDBY); */
+		/* Hier teilen wir der Time.sh mit das die Box Standby ist */
+		system("echo scart > /proc/stb/avs/0/input");
 		
 		//if(videoDecoder)
 		//	videoDecoder->SetInput(INPUT_SCART);
@@ -4453,6 +4456,7 @@ void CNeutrinoApp::standbyMode( bool bOn
 		// video wake up
 		//if(videoDecoder)
 		//	videoDecoder->SetInput(INPUT_ENCODER);
+		system("echo encoder > /proc/stb/avs/0/input");
 
 		g_Sectionsd->setPauseScanning(false);
 		g_Sectionsd->setServiceChanged(live_channel_id&0xFFFFFFFFFFFFULL, true );
--- neutrino-hd2-exp.org/src/gui/Makefile.am	2013-06-13 18:25:05.000000000 +0200
+++ neutrino-hd2-exp/src/gui/Makefile.am	2013-06-18 18:52:11.000000000 +0200
@@ -8,6 +8,9 @@ sleep 180 && rm svn_version.h
 	else \
 		rm svn_version.h; echo '#define BUILT_DATE "'`date`'"' > svn_version.h ; \
 	fi
+teamcs_sources =teamcs.cpp teamcsemu.cpp teamcsuhr.cpp teamcswlan.cpp teamcssystem.cpp \
+		teamcstuner.cpp teamcsfb.cpp teamcsupnp.cpp teamcssamba.cpp teamcsswap.cpp \
+		teamcsoverclock.cpp teamcsmac.cpp
 
 noinst_HEADERS = svn_version.h
 
@@ -55,7 +55,8 @@ libneutrino_gui_a_SOURCES = \
 
 libneutrino_gui2_a_SOURCES = \
 	color.cpp motorcontrol.cpp hdd_menu.cpp \
-	alphasetup.cpp dvbsub_select.cpp psisetup.cpp opkg_manager.cpp webtv.cpp
+	alphasetup.cpp dvbsub_select.cpp psisetup.cpp opkg_manager.cpp webtv.cpp \
+	$(teamcs_sources)
 
 if ENABLE_UPNP
 libneutrino_gui2_a_SOURCES += upnpbrowser.cpp
--- neutrino-hd2-exp.org/src/neutrino_menue.cpp	2013-06-18 16:45:07.000000000 +0200
+++ neutrino-hd2-exp/src/neutrino_menue.cpp	2013-06-18 18:52:11.000000000 +0200
@@ -114,6 +114,8 @@
 #include "gui/upnpbrowser.h"
 #endif
 
+#include "gui/teamcs.h"
+
 #include <system/setting_helpers.h>
 #include <system/settings.h>
 #include <system/debug.h>
@@ -266,6 +268,14 @@ void CNeutrinoApp::InitMainMenu(CMenuWid
 	MediaPlayer.addItem(new CMenuForwarderItemMenuIcon(LOCALE_UPNPBROWSER_HEAD, true, NULL, new CUpnpBrowserGui(), NULL, CRCInput::RC_blue, NEUTRINO_ICON_BUTTON_BLUE, "upnpbrowser", LOCALE_HELPTEXT_UPNPBROWSER ));
 #endif
 
+       //teamcs
+       /*mainMenu.addItem(new CMenuForwarderItemMenuIcon(LOCALE_MAINMENU_TEAMCS, true, "", this, "teamcs", CRCInput::convertDigitToKey(shortcut++), NULL, "teamcs", LOCALE_HELPTEXT_TEAMCS ));*/
+       mainMenu.addItem(new CMenuForwarderItemMenuIcon(LOCALE_MAINMENU_TEAMCS, true, "", new teamcs(), NULL, CRCInput::convertDigitToKey(shortcut++), NULL, "teamcs", LOCALE_HELPTEXT_TEAMCS ));
+
+       #if ENABLE_GRAPHLCD
+        mainMenu.addItem(new CMenuForwarderItemMenuIcon(LOCALE_MAINMENU_GLCD1, true, "", new GLCD_Menu(), NULL, CRCInput::convertDigitToKey(shortcut++), NULL, "nix", LOCALE_HELPTEXT_NIX ));
+       #endif
+
 	// vcr-scart
 #if defined (ENABLE_SCART)
 	mainMenu.addItem(new CMenuForwarderItemMenuIcon(LOCALE_MAINMENU_SCARTMODE, true, NULL, this, "scart", CRCInput::RC_blue, NEUTRINO_ICON_BUTTON_BLUE, "scart", LOCALE_HELPTEXT_SCART ));
