--- stm-target-gcc-orig.spec	2013-04-13 18:54:10.984915000 +0200
+++ stm-target-gcc.spec	2013-04-13 20:05:20.534087491 +0200
@@ -4,11 +4,11 @@
 Version: 4.6.3
 %else
 %if_target_cpu arm
-Version: 4.6.3
-%else
+
+
 Version: 4.2.4
 %endif
-%endif
+
 
 Release: 115
 
@@ -306,14 +306,14 @@
 %if_target_arch_no_uclibc
         --with-tls=gnu2 \
 %endif
-%endif
+
 %if "%{_stm_target_cpu_id}" != "%%%%{_stm_target_cpu_id}"
 	--with-cpu=%{_stm_target_cpu_id} \
 %endif
 %if "%{_stm_target_gcc_config}" != "%%%%{_stm_target_gcc_config}"
 	%{_stm_target_gcc_config} \
 %endif
-        && true
+         true
 
 # This package is NOT pmake safe
 %pmake
@@ -520,7 +520,7 @@
 %{_stm_cross_target_dir}%{_stm_target_lib_dir}/libgomp.spec
 %{_stm_cross_target_dir}/%{_stm_target_info_dir}/libgomp.info.gz
 %endif
-%endif
+
 
 %{_gccdir}/include/stdfix.h
 %{_gccdir}/include/stdint-gcc.h
