--- enigma2-nightly.org/lib/driver/vfd.cpp	2013-03-19 19:49:07.000000000 +0100
+++ enigma2-nightly/lib/driver/vfd.cpp	2013-03-23 11:18:50.958408322 +0100
@@ -186,7 +186,7 @@ void * start_loop (void *arg)
 	evfd vfd;
 	blocked = true;
 	//vfd.vfd_clear_icons();
-	vfd.vfd_write_string("AAF DEV-TEAM", true);
+	vfd.vfd_write_string("Lade...", true);
 	//run 2 times through all icons 
 	for  (int vloop = 0; vloop < 128; vloop++) {
 #if !defined(PLATFORM_FORTIS_HDBOX) && !defined(PLATFORM_OCTAGON1008) && !defined(PLATFORM_ATEVIO7500) && !defined(PLATFORM_CUBEREVO) && !defined(PLATFORM_CUBEREVO_MINI) && !defined(PLATFORM_CUBEREVO_MINI2) && !defined(PLATFORM_CUBEREVO_MINI_FTA) && !defined(PLATFORM_CUBEREVO_250HD) && !defined(PLATFORM_CUBEREVO_2000HD) && !defined(PLATFORM_CUBEREVO_9500HD)
